eclair-core/src/main/scala/fr/acinq/eclair/payment/relay/NodeRelayer.scala

```diff
@@ -165,17 +166,30 @@ class NodeRelayer(nodeParams: NodeParams, relayer: ActorRef, router: ActorRef, c
 
   private def rejectHtlc(htlcId: Long, channelId: ByteVector32, amount: MilliSatoshi, failure: Option[FailureMessage] = None): Unit = {
     val failureMessage = failure.getOrElse(IncorrectOrUnknownPaymentDetails(amount, nodeParams.currentBlockHeight))
-    commandBuffer ! CommandBuffer.CommandSend(channelId, htlcId, CMD_FAIL_HTLC(htlcId, Right(failureMessage), commit = true))
+    commandBuffer ! CommandBuffer.CommandSend(channelId, CMD_FAIL_HTLC(htlcId, Right(failureMessage), commit = true))
   }
 
   private def rejectPayment(upstream: Upstream.TrampolineRelayed, failure: Option[FailureMessage] = None): Unit =
     upstream.adds.foreach(add => rejectHtlc(add.id, add.channelId, upstream.amountIn, failure))
 
   private def fulfillPayment(upstream: Upstream.TrampolineRelayed, paymentPreimage: ByteVector32): Unit = upstream.adds.foreach(add => {
     val cmdFulfill = CMD_FULFILL_HTLC(add.id, paymentPreimage, commit = true)
-    commandBuffer ! CommandBuffer.CommandSend(add.channelId, add.id, cmdFulfill)
+    commandBuffer ! CommandBuffer.CommandSend(add.channelId, cmdFulfill)
   })
 
+  override def mdc(currentMessage: Any): MDC = {
+    val paymentHash_opt = currentMessage match {
+      case IncomingPacket.NodeRelayPacket(add, _, _, _) => Some(add.paymentHash)
```

Comment:
nit: using extractors is nice but it will break everytime the signature of any of the classes changes